src/components/implementation/tests/micro_booter/micro_booter.c

```diff
@@ -568,7 +568,7 @@ term_fn(void *d)
 void
 cos_init(void)
 {
-	cos_meminfo_init(&booter_info.mi, BOOT_MEM_KM_BASE, COS_MEM_KERN_PA_SZ);
+	cos_meminfo_init(&booter_info.local_mi, BOOT_MEM_KM_BASE, COS_MEM_KERN_PA_SZ, BOOT_CAPTBL_SELF_LOCAL_PT);
```

Comment:
1. that means that there are ton of modifications in this PR that make it very hard to go through for almost no gain.

2. That would make your improvements very limited in their intended use.  What if there are other devices that expose physical memory but don't expose global memory?  Xeon Phis are an example.

You're providing a general facility for using multiple physical page-tables for allocation.  Why bind it one one use-case?